### PR TITLE
[feat]: [CDS-72708]: separate command unit for service and overrides

### PR DESCRIPTION
**Pull request by tchaurasiya**
## Describe your changes

Screenshot 2023-07-25 at 1 42 30 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49916)

### CODE REVIEW

**Comment by tchaurasiya** [comment]
trigger uts

**Comment by tchaurasiya** [comment]
Keeping this pr open, as it required another pipeline service pr to be deployed in prod.
https://github.com/harness/harness-core/pull/49919

**Comment by tchaurasiya** [comment]
Pipeline service has been deployed to prod.
Getting review and we will merge this pr

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepV3.java [hunk 1]
```diff
@@ -710,12 +723,25 @@ public StepResponse handleChildrenResponse(
...
+    final NGLogCallback logCallback =
+        serviceStepsHelper.getServiceLogCallback(ambiance, false, SERVICE_STEP_COMMAND_UNIT);
     if (StatusUtils.brokeStatuses().contains(stepResponse.getStatus())) {
       saveExecutionLog(logCallback, LogHelper.color("Failed to complete service step", LogColor.Red), LogLevel.INFO,
           CommandExecutionStatus.FAILURE);
+      serviceStepUnitProgress = UnitProgress.newBuilder()
+                                    .setStatus(UnitStatus.FAILURE)
+                                    .setUnitName(SERVICE_STEP_COMMAND_UNIT)
+                                    .setStartTime(System.currentTimeMillis())
+                                    .setEndTime(System.currentTimeMillis())
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepV3.java:735]: start and end time are same here ?

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/steps/ArtifactsStepV2.java [hunk 2]
```diff
@@ -180,7 +181,8 @@ public AsyncExecutableResponse executeAsyncAfterRbac(
 
     final ArtifactListConfig artifacts = service.getServiceDefinition().getServiceSpec().getArtifacts();
 
-    final NGLogCallback logCallback = serviceStepsHelper.getServiceLogCallback(ambiance);
+    final NGLogCallback logCallback =
+        serviceStepsHelper.getServiceLogCallback(ambiance, false, SERVICE_STEP_COMMAND_UNIT);
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/steps/ArtifactsStepV2.java:185]: I think that the 
```java
serviceStepsHelper.getServiceLogCallback() should default to SERVICE_STEP_COMMAND_UNIT itself
```

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/ServiceOverrideUtilityFacade.java [hunk 3]
```diff
@@ -180,40 +191,112 @@ private EnumMap<ServiceOverridesType, NGServiceOverrideConfigV2> getMergedOverri
...
+    overrideLogCallback.saveExecutionLog(
+        "Successfully collected overrides.", LogLevel.INFO, CommandExecutionStatus.SUCCESS);
 
     return overridesForStep;
   }
 
+  private void logOverridesDetails(Map<Scope, NGServiceOverridesEntity> infraServiceOverride,
+      Map<Scope, NGServiceOverridesEntity> infraOverride, Map<Scope, NGServiceOverridesEntity> envServiceOverride,
+      Map<Scope, NGServiceOverridesEntity> envOverride, NGLogCallback overrideLogCallback) {
+    if (isNotEmpty(infraServiceOverride)) {
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/ServiceOverrideUtilityFacade.java:240]: should we try/catch here on the log part ?

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/steps/ArtifactsStepV2.java (hunk 2 above)
Comment (tchaurasiya) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/steps/ArtifactsStepV2.java:185]: Yes Yogesh, both are working.
but explicitly providing command unit looks more clear.

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/ServiceOverrideUtilityFacade.java (hunk 3 above)
Comment (tchaurasiya) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/ServiceOverrideUtilityFacade.java:240]: done

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepV3.java (hunk 1 above)
Comment (tchaurasiya) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/ServiceStepV3.java:735]: fixed.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 6 of 11 files at r1, 1 of 1 files at r3, 4 of 5 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49916)* status: 12 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Comment by tchaurasiya** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49916)* status: 12 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by lovish1999 — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 3 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49916)* status: 15 file(s) to review. Waiting on team(s): del-team.

**Comment by yogesh-chauhan** [comment]
trigger review

**Pull request merged**